### PR TITLE
10979 and 1960 change reset to 1 yr, add qb department to txns and division

### PR DESCRIPTION
**Pull request by DevneyHamilton**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,6 +36,8 @@ coverage/
 frpc.ini
 ~$*
 frpc.ini
+app/views/bin
+bin/update
```
Comment (DevneyHamilton) [inline .gitignore]: Is this ok?

File: app/models/accounting/qb/department.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#
+# Represents a QB Department object and can create a reference object
+# for a link to this object in a transaction or other QB object.
+
+class Accounting::QB::Department < ApplicationRecord
+  QB_OBJECT_TYPE = 'Department'
+
+  belongs_to :division, optional: true, class_name: "Division"
+
+  def self.create_or_update_from_qb_object!(qb_object_type:, qb_object:)
```
[the remark this answers is not in the document]
Comment (DevneyHamilton) [reply app/models/accounting/qb/department.rb:33]: need this to be consistent with usage in base fetcher

**Comment by DevneyHamilton** [comment]
This PR may be a little messy to review. The main changes are:
- change data reset time to 1 yr instead of 3 months (10979)
- change Quickbooks namespace to QB
- import QB Departments, associate them with divisions, and use a loan's division's department when creating a transaction. The QB Departments are called QB divisions in user-facing language because that is what TWW staff refer to them as, and QB also calls departments 'divisions' in the UI. 

The UI for the department/division changes is:

setting qb division on Madeline division: 
Screen Shot 2020-08-26 at 4 09 25 PM

notice that qb division is not set:
Screen Shot 2020-08-26 at 1 17 02 PM

here is the division in the QB view of the txn:
Screen Shot 2020-08-26 at 1 21 27 PM

The QB division is not visible elsewhere in the txn experience. We could update that here or in future issues.

File: app/controllers/admin/divisions_controller.rb [hunk 3]
```diff
@@ -51,8 +51,11 @@ def new
 
   def update
     authorize @division
-
-    if @division.update(division_params)
+    division_params = prep_division_params
+    qb_department_id = division_params.delete(:qb_department_id)
+    @division.update(division_params)
```
Comment (meltheadorable) [inline app/controllers/admin/divisions_controller.rb:56]: I think all this logic related to manipulating and fetching the department id should go into `prep_division_params`

File: .gitignore (hunk 1 above)
Comment (meltheadorable) [reply .gitignore]: Why the change here?

**Review by meltheadorable — APPROVED** [review]
Looks good mostly

**Pull request merged**